src/codegen/insert/insert_helpers.cpp

```diff
@@ -0,0 +1,84 @@
+//===----------------------------------------------------------------------===//
+//
+//                         Peloton
+//
+// insert_helpers.cpp
+//
+// Identification: src/codegen/insert/insert_helpers.cpp
+//
+// Copyright (c) 2015-17, Carnegie Mellon University Database Group
+//
+//===----------------------------------------------------------------------===//
+
+#include "concurrency/transaction_manager_factory.h"
+#include "codegen/insert/insert_helpers.h"
+#include "storage/tuple.h"
+
+namespace peloton {
+namespace codegen {
+
+bool InsertHelpers::InsertRawTuple(concurrency::Transaction *txn,
+                                   storage::DataTable *table,
+                                   const storage::Tuple *tuple) {
+
+  concurrency::TransactionManager *txn_mgr =
+      &concurrency::TransactionManagerFactory::GetInstance();
+
+  // Insert tuple into the table.
+  // Tuple will be copied.
+  ItemPointer *index_entry_ptr = nullptr;
+  ItemPointer location = table->InsertTuple(tuple, txn, &index_entry_ptr);
+
+  // It is possible that some concurrent transactions have inserted the same
+  // tuple. In this case, abort the transaction.
+  if (location.block == INVALID_OID) {
+    txn_mgr->SetTransactionResult(txn, ResultType::FAILURE);
+    return false;
+  }
+
+  txn_mgr->PerformInsert(txn, location, index_entry_ptr);
+
+  return true;
+}
+
+void InsertHelpers::InsertValue(concurrency::Transaction *txn,
+                                storage::DataTable *table,
+                                char *value, size_t num_tuples) {
```

Comment:
_num_tuples_ seems a total size of the tuples to me.  Can we change the variable name?